src/main/scala/diplomacy/Nodes.scala

```diff
@@ -1261,6 +1262,7 @@ sealed abstract class MixedNode[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data](
     * and return all the dangles of this node.
     */
   protected[diplomacy] def instantiate(): Seq[Dangle] = {
+    instantiated = true
     bundlesSafeNow = true
```

Comment:
As I mentioned above, it seems like these are exactly the same thing with the caveat that `bundlesSafeNow` is turned back off again in `finishInstantiate` below. But... it seems that code as originally written had the design assumption that the only time in Scala execution that one would want to manipulate such a bundle is during the execution of its `LazyModuleImp`. Your use case is literally disproving that claim, and I can imagine similar issues happening with e.g. `BoringUtils` or any other spooky-action-at-a-distance libraries. So I think we should retire `bundlesSafeNow` in favor of your `instantiated`, which should preserve the guard against accessing bundles too early while allowing them to be accessed "too late".

Reply:
Might be good to factor that change out into a separate PR though.